server/fleet/windows_mdm_test.go

```diff
@@ -331,23 +395,147 @@ func TestValidateUserProvided(t *testing.T) {
 </Replace>
 `),
 			},
-			wantErr: false,
+			wantErr: "",
 		},
 		{
 			name: "Valid XML with reserved name",
 			profile: MDMWindowsConfigProfile{
 				Name:   mdm.FleetWindowsOSUpdatesProfileName,
 				SyncML: []byte(`<Replace><Target><LocURI>Custom/URI</LocURI></Target></Replace>`),
 			},
-			wantErr: true,
+			wantErr: `Profile name "Windows OS Updates" is not allowed`,
+		},
+		{
+			name: "XML with top level comment",
+			profile: MDMWindowsConfigProfile{
+				SyncML: []byte(`
+				  <!-- this is a comment -->
+				  <Replace>
+				    <Target>
+				      <LocURI>Custom/URI</LocURI>
+				    </Target>
+				  </Replace>
+				`),
+			},
+			wantErr: "",
+		},
+		{
+			name: "XML with nested root element in data",
+			profile: MDMWindowsConfigProfile{
+				SyncML: []byte(`
+				  <Replace>
+				    <Item>
+				    <Target>
+				      <LocURI>Custom/URI</LocURI>
+				      <Data>
+				        <?xml version="1.0"?>
+					<Foo></Foo>
+				      </Data>
+				    </Target>
+				    </Item>
+				  </Replace>
+				`),
+			},
+			wantErr: "The file should include valid XML",
+		},
+		{
+			name: "XML with nested root element under Replace",
+			profile: MDMWindowsConfigProfile{
+				SyncML: []byte(`
+				  <Replace>
+				    <?xml version="1.0"?>
+				    <Item>
+				    <Target>
+				      <LocURI>Custom/URI</LocURI>
+				      <Data>
+					<Foo></Foo>
+				      </Data>
+				    </Target>
+				    </Item>
+				  </Replace>
+				`),
+			},
+			wantErr: "The file should include valid XML",
+		},
+		{
+			name: "XML with root element above Replace",
+			profile: MDMWindowsConfigProfile{
+				SyncML: []byte(`
+				  <?xml version="1.0"?>
+				  <Replace>
+				  <Item>
+				    <Target>
+				      <LocURI>Custom/URI</LocURI>
+				      <Data>
+					<Foo></Foo>
+				      </Data>
+				    </Target>
+				    </Item>
+				  </Replace>
+				`),
+			},
+			wantErr: "The file should include valid XML",
+		},
+		{
+			name: "XML with root element inside Target",
+			profile: MDMWindowsConfigProfile{
+				SyncML: []byte(`
+				  <Replace>
+				    <Target>
+				      <?xml version="1.0"?>
+				      <LocURI>Custom/URI</LocURI>
+				      <Data>
+					<Foo></Foo>
+				      </Data>
+				    </Target>
+				  </Replace>
+				`),
+			},
+			wantErr: "The file should include valid XML",
+		},
+		{
+			name: "XML escaped with nested root element",
```

Comment:
This seems to be the `CDATA` case the escaped XML. Is this name mixed up with the name of the next case?